app/models/bike.rb

```diff
@@ -809,30 +824,12 @@ def propulsion_type_name
     PropulsionType.new(propulsion_type).name
   end
 
-  # Geolocate based on the full current stolen record address, if available.
-  # Otherwise, use the data set by set_location_info.
-  def geocode_data
-    return @geocode_data if defined?(@geocode_data)
-
-    # Sets lat/long, will avoid a geocode API call if coordinates are found
-    set_location_info
-
-    @geocode_data =
-      current_stolen_record
-        &.address(force_show_address: true)
-        .presence ||
-      [city, zipcode, country&.name]
-        .select(&:present?)
-        .join(" ")
-        .presence
-  end
-
   # Take lat/long from associated geocoded model
-  # Only geocode if no lat/long present and geocode data present
+  # Only geocode if lat/long are blank and address present
   def should_be_geocoded?
     return false if skip_geocoding?
     return false if latitude.present? && longitude.present?
-    geocode_data.present?
+    address.present?
```

Comment:
I'm a little confused by this, it seems circular - but maybe I'm missing something?

- Before validation check this method (`should_be_geocoded?`)
- This line checks if address is present
- If address *is* present, we set the address.

Reply:
Good call-out—it's actually L830 that's operative in that situation (in tests) but using `:should_be_geocoded?` muddies that a bit. Fixed in 8d6205df